lib/galaxy/tools/execute.py

```diff
@@ -91,6 +96,7 @@ def execute_single_job(execution_slice, completed_job):
     jobs_executed = 0
     has_remaining_jobs = False
     execution_slice = None
+    job_datasets = {}  # job: list of dataset instances created by job
```

Comment:
@jmchilton  Do you mean why it's not initialized inside the `execute_single_job` function, after line 73? 
We init it on line 99, after that `execute_single_job` may be called (line 106) where `job_datasets` may be populated (line 76), after which, back in the `execute` method it may be iterated over (line 118). 
Sorry if I misunderstood the question.